### PR TITLE
No weird tree sitter output

### LINKED ISSUE #1543: JSONDecodeError in 0.20
```
$ semgrep -f ../njsscan/njsscan/rules/semantic_grep/ ../njsscan/tests/assets/templates/
running 107 rules...
100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████|107/107
(venv) CA-AJINA-M:njsscan ajinabraham$ cd Code/libsast/
(venv) CA-AJINA-M:njsscan ajinabraham$ libsast -s ../njsscan/njsscan/rules/semantic_grep/ ../njsscan/tests/assets/templates/
Traceback (most recent call last):
  File "/Code/njsscan/venv/bin/libsast", line 11, in <module>
    load_entry_point('libsast==1.1.6', 'console_scripts', 'libsast')()
  File "/Code/njsscan/venv/lib/python3.7/site-packages/libsast/__main__.py", line 93, in main
    result = Scanner(options, args.path).scan()
  File "/Code/njsscan/venv/lib/python3.7/site-packages/libsast/scanner.py", line 56, in scan
    self.options).scan(valid_paths)
  File "/Code/njsscan/venv/lib/python3.7/site-packages/libsast/core_sgrep/semantic_sgrep.py", line 36, in scan
    sgrep_out = invoke_semgrep(paths, self.scan_rules)
  File "/Code/njsscan/venv/lib/python3.7/site-packages/libsast/core_sgrep/helpers.py", line 41, in invoke_semgrep
    **kwargs,
  File "/Code/njsscan/venv/lib/python3.7/site-packages/semgrep/semgrep_main.py", line 362, in main
    ).invoke_semgrep(target_manager, all_rules)
  File "/Code/njsscan/venv/lib/python3.7/site-packages/semgrep/core_runner.py", line 490, in invoke_semgrep
    rules, target_manager
  File "/Code/njsscan/venv/lib/python3.7/site-packages/semgrep/core_runner.py", line 460, in _run_rules
    rule, target_manager, semgrep_core_ast_cache_dir, max_timeout_files
  File "/Code/njsscan/venv/lib/python3.7/site-packages/semgrep/core_runner.py", line 372, in _run_rule
    cache_dir,
  File "/Code/njsscan/venv/lib/python3.7/site-packages/semgrep/core_runner.py", line 299, in _run_core_command
    output_json = json.loads((core_run.stdout.decode("utf-8", "replace")))
  File "/env/.pyenv/versions/3.7.7/lib/python3.7/json/__init__.py", line 348, in loads
    return _default_decoder.decode(s)
  File "/env/.pyenv/versions/3.7.7/lib/python3.7/json/decoder.py", line 337, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
  File "/env/.pyenv/versions/3.7.7/lib/python3.7/json/decoder.py", line 355, in raw_decode
    raise JSONDecodeError("Expecting value", s, err.value) from None
json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
```

This is how I am calling semgrep https://github.com/ajinabraham/libsast/blob/master/libsast/core_sgrep/helpers.py#L9

### PR DESCRIPTION
**Pull request by nbrahms**
Fixes #1543.

Trio-programmed with @colleend @mjambon

### CODE REVIEW

File: semgrep-core/parsing/Parse_javascript_tree_sitter.ml [hunk 1]
```diff
@@ -1758,10 +1758,13 @@ let parse file =
     program env ast
   with
     (Failure "not implemented") as exn ->
-      let s = Printexc.get_backtrace () in
-      pr2 "Some constructs are not handled yet";
-      pr2 "CST was:";
-      CST.dump_tree ast;
-      pr2 "Original backtrace:";
-      pr2 s;
+      (* This debugging output is not JSON and breaks core output
```
Comment (colleend) [inline semgrep-core/parsing/Parse_javascript_tree_sitter.ml:1761]: do we need to comment this out?

File: semgrep-core/parsing/Parse_code.ml [hunk 2]
```diff
@@ -76,7 +78,7 @@ let just_parse_with_lang lang file =
               Ast_js_build.program cst
             )
             file
-        with _exn -> Parse_javascript_tree_sitter.parse file
+        with ForceTreeSitter -> Parse_javascript_tree_sitter.parse file
```
Comment (aryx) [inline semgrep-core/parsing/Parse_code.ml:81]: We want to try treesitter if there is any exn in the pfff parser (lexer exn, parse error) or when we have -forcetreesitter

Comment (aryx) [reply semgrep-core/parsing/Parse_code.ml:81]: So it should remain _exn here

File: semgrep-core/parsing/Parse_javascript_tree_sitter.ml (hunk 1 above)
Comment (aryx) [reply semgrep-core/parsing/Parse_javascript_tree_sitter.ml:1761]: But it is printed on stderr. It should not matter. The json is printed on stdout

**Review by aryx — CHANGES_REQUESTED** [review]

File: semgrep-core/parsing/Parse_javascript_tree_sitter.ml (hunk 1 above)
Comment (nbrahms) [reply semgrep-core/parsing/Parse_javascript_tree_sitter.ml:1761]: Commenting it out fixes the bug, so 🤷

File: semgrep-core/parsing/Parse_code.ml (hunk 2 above)
Comment (nbrahms) [reply semgrep-core/parsing/Parse_code.ml:81]: You and @mjambon disagree here...

**Review by aryx — APPROVED** [review]
Ok let merge for now

File: semgrep-core/parsing/Parse_code.ml (hunk 2 above)
Comment (mjambon) [reply semgrep-core/parsing/Parse_code.ml:81]: I suggest being explicit about the two meanings here:
* is it a force_tree_sitter request?
* is it an error and we're falling back to the tree-sitter parser?

I also suggest printing any exception caught with a catch-all (`_exn`) so we know what went wrong.

File: semgrep-core/parsing/Parse_javascript_tree_sitter.ml (hunk 1 above)
Comment (mjambon) [reply semgrep-core/parsing/Parse_javascript_tree_sitter.ml:1761]: @aryx when you get back to it, you probably should check if by any chance semgrep merges semgrep-core's stderr into stdout.

File: semgrep-core/parsing/Parse_code.ml (hunk 2 above)
Comment (aryx) [reply semgrep-core/parsing/Parse_code.ml:81]: The idea is to not print what went wrong when the error is a parse error in the pfff parser.
We should not bother the user with that. We first try with pfff, and if it fails we try with tree-sitter (or the reverse, it depends on the language).
But I'll change the option to -tree_sitter_only, maybe it's clearer.

**Pull request merged**